eq-author/src/apollo/resolvers/index.js

```diff
@@ -0,0 +1,125 @@
+import gql from "graphql-tag";
+
+export const typeDefs = gql`
+  extend type QuestionPage {
+    isNew: Boolean!
+  }
+
+  extend type Mutation {
+    createQuestionPage: QuestionPage
+    updateQuestionPage: QuestionPage
+    duplicatePage: QuestionPage
+  }
+`;
+
+const GET_NEW_PAGES_VISITED_LIST = gql`
+  query newPagesVisitedList {
+    newPagesList @client
+    pagesVisitedList @client {
+      id
+      firstVisit
+    }
+  }
+`;
+
+const initialiseMultiplePages = (pages, cache) => {
+  const { newPagesList } = cache.readQuery({
+    query: GET_NEW_PAGES_VISITED_LIST,
+  });
+
+  pages.forEach(questionPage => {
+    newPagesList.push(questionPage.id);
+  });
+
+  cache.writeData({ data: { newPagesList } });
+};
+
+const isNewQuestionPage = (questionPage, args, { cache }) => {
+  const { newPagesList, pagesVisitedList } = cache.readQuery({
```

Comment:
I'm not sure I understand why we need to track the pages visited as well?

Reply:
It was a workaround because new QuestionPage was returned twice - once for createQuestionPage mutation and again for subsequent getPage query. Refactored so that new page calls update mutation regardless of whether no values have changed. Now only need to track if page is new